doc: update on the programming firmware section of Thingy 91 (#8720)
nrfconnect/sdk-nrf: doc/nrf/ug_thingy91_gsg.rst

Before you start using the Thingy:91, it is recommended that you update the application firmware to :ref:`asset_tracker_v2`.
You must also update the modem firmware.
You can update the Thingy:91 application and modem firmware through the :term:`Universal Serial Bus (USB)` cable using MCUboot.
The MCUboot is a secure bootloader used to update applications in case you do not have an external debugger.

> Alternatively, you can use an external debug probe.

Reviewer: It might make sense to move the note about the debug probe  that is further down (https://github.com/nrfconnect/sdk-nrf/pull/8720/files#diff-90d0e4bb792f25c3e98c1eeb33e528b8d7ff1ea2413b4975771ffeb64ec4fcd0R102) up here.

> Alternatively, you can use an external debug probe.

Reviewer: @gregersrygg, any updates on the recommended order after we discussed it offline?

> Alternatively, you can use an external debug probe.

Reviewer 2: I had a chat with the product owner now. Since the thingy:91 getting started guide focuses on programming pre-compiled firmware he wanted to keep focus on the MCUboot programming in that guide.

However, in the «Developing with thingy:91» guide, he thinks  we should state that it's recommended to use an external debugger (does not have to be in this PR).

Also, the [list of hardware](https://developer.nordicsemi.com/nRF_Connect_SDK/doc/latest/nrf/ug_thingy91_gsg.html#hardware) in the getting started guide could list an external debugger (or 91DK) and the debug cable as recommended if planning to develop on the thingy:91. Since new customers might use that list to plan what hardware they need for developing a prototype (does not have to be in this PR).

> Alternatively, you can use an external debug probe.

Reviewer: Thanks for checking!

> Alternatively, you can use an external debug probe.

Author: Thanks @gregersrygg . I will create a new separate PR to cover the 2 points you mentioned above.

> The board enters bootloader mode if any of the following buttons are pressed after the Thingy:91 is powered (using **SW1**):

Reviewer 3: ```suggestion
The board enters bootloader mode if any of the following buttons are pressed while the Thingy:91 is being powered (using **SW1**):
```

> The board enters bootloader mode if any of the following buttons are pressed while the Thingy:91 is being powered (using **SW1**):

Reviewer: ```suggestion
The board enters bootloader mode if any of the following buttons are pressed while the Thingy:91 is being powered on (using **SW1**):
```

> * **SW3** - The main button used to flash the nRF9160.

Reviewer: ```suggestion
* **SW3** - The main button used to flash the nRF9160 SiP.
```

> * **SW4** - The button used to update the nRF52840.

Reviewer: ```suggestion
* **SW4** - The button used to update the nRF52840 SoC.
```
